bitcoin_executor/README.md

```diff
@@ -0,0 +1,35 @@
+![Logo!](assets/logo.jpg)
+
+# Bitcoin Executor
+
+Bitcoin node implementation powered by Sui smart contract.
+
+## Components
+
+1. Bitcoin [Relayer](https://github.com/gonative-cc/relayer/tree/master/bitcoinspv): Go service listening for new blocks, validating and sending to SPV, Walrus and Executor
+2. Bitcoin [Executor](./) (this project): manages UTXO state and executes Bitcoin transactions.
+3. [Walrus](https://www.walrus.xyz/): data availability for Bitcoin blocks.
+4. Bitcoin [SPV](https://github.com/gonative-cc/move-bitcoin-spv/): light client validating Bitcoin block headers, checking proof of work and managing the heaviest chain.
+
+## Flow
+
+1. Relayer listens for _new blocks_.
+2. Once a _new block_ arrives, Relayer validates it and checks if it's already processed. If not valid or already processed, `return`.
+   - NOTE: Relayer has internal storage to handle reorgs and query Bitcoin nodes for missing blocks of the _current branch_.
+3. Relayer sends block header (or a whole branch if reorg is detected) to SPV to validate if the branch makes a heaviest chain. If not, `return`.
+4. Relayer sends new blocks of the extended branch (based on the SPV result) Walrus for data availability.
+5. Relayer sends new blocks to Executor.
+6. Executor parses the block checks if it was processed, if yes, `return`.
```

Comment:
Executor doesnt check if a block was processed, rather it checks if the `UTXO` the tx in the block is trying to spend exist and has not been spend before.

Reply:
OK, updated